test_runner/regress/test_compatibility.py

```diff
@@ -177,23 +169,16 @@ def test_backward_compatibility(
     )
 
     try:
-        # Copy the snapshot to current directory, and prepare for the test
-        prepare_snapshot(
-            from_dir=compatibility_snapshot_dir,
-            to_dir=test_output_dir / "compatibility_snapshot",
-            port_distributor=port_distributor,
-        )
+        neon_env_builder.num_safekeepers = 3
+        neon_env_builder.enable_generations = False
```

Comment:
I'm wondering what breaks if we don't set enable_generations = False?

Reply:
Haha, I had this question in mind for future us, but since you've asked:
I've had some failures when a pageserver deleted tenants, I don't remember the exact message. But now I started to doubt if my configuration was correct. Pushing 598f90933a3e6c63d6ae9b9190f944fa44db3022 to check the exact error.

Reply:
Here it is:

- `test_forward_compatibility` (old binaries, new data) ([link](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6087/7170803255/index.html#suites/0fbaeb107ef328d03993d44a1fb15690/b9f1769a0db51499/?attachment=f99f09fe52045199))
```
2023-12-11T18:21:32.228653Z  INFO loading tenant configuration from /tmp/test_output/test_forward_compatibility[debug-pg14]/repo/pageserver_1/tenants/39424a9cef59b97cd2bd8c98edb1a62d/config-v1
2023-12-11T18:21:32.233007Z  INFO init_tenant_mgr: Calling control plane API to re-attach tenants
2023-12-11T18:21:32.235651Z  INFO init_tenant_mgr: Received re-attach response with 0 tenants
2023-12-11T18:21:32.236614Z  INFO init_tenant_mgr: Detaching tenant, control plane omitted it in re-attach response tenant_id=39424a9cef59b97cd2bd8c98edb1a62d shard_id=0000
2023-12-11T18:21:32.237341Z  INFO deletion frontend: Replay complete next_sequence=1
```

- `test_backward_compatibility` (new binaries, old data) ([link](https://neon-github-public-dev.s3.amazonaws.com/reports/pr-6087/7170803255/index.html#suites/0fbaeb107ef328d03993d44a1fb15690/15bda75ed5f55fb5/?attachment=420445ef5c31d0e1))
```
2023-12-11T18:21:24.539301Z  INFO loading tenant configuration from /tmp/test_output/test_backward_compatibility[debug-pg14]/repo/pageserver_1/tenants/491f531edd30f22d0591d3151587b94b/config
2023-12-11T18:21:24.546655Z  INFO init_tenant_mgr: Calling control plane API to re-attach tenants
2023-12-11T18:21:24.549583Z  INFO init_tenant_mgr: Received re-attach response with 0 tenants
2023-12-11T18:21:24.550294Z  INFO init_tenant_mgr: Detaching tenant, control plane omitted it in re-attach response tenant_id=491f531edd30f22d0591d3151587b94b shard_id=0000
2023-12-11T18:21:24.552061Z  INFO deletion frontend: Replay complete next_sequence=1
```

@jcsp is there anything we can do despite disabling generations?

Reply:
A-ha, thanks for the info!
We already collect snapshots with `attachments.json`, I've added d3d8b4cb74760b2cfa71f4418f3ca142558116fb to also restore this file